src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,45 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents a command to add a task.
+ */
+public class AddCommand extends Command {
+    /**
+     * * Task to be added.
+     */
+    private Task taskToAdd;
+
+    /**
+     * Constructor for AddCommand.
+     *
+     * @param taskToAdd the task to be added
+     */
+    public AddCommand(Task taskToAdd) {
+        super();
+        this.taskToAdd = taskToAdd;
+    }
+
+    /**
+     * Executes the command to add a task.
+     * It adds the task to the TaskList provided by calling addTaskToList from
+     * the TaskList class.
+     *
+     * @param tasks   The TaskList to be searched
+     * @param ui      The ui to send messages to
+     * @param storage Storage if required for the execution.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        tasks.addTaskToList(this.taskToAdd);
+        ui.messageUser("Got it. I've added this task:",
```

Comment:
Maybe the job of what messages to show the user can be handled by Ui directly instead, where Ui has corresponding methods for the different executions which can be called here. This might be helpful if the format of messages in the program change, where it can edited in one file instead of across different files.